src/spinner.js

```diff
@@ -1,31 +1,31 @@
-import React from 'react';
-import { connect } from 'react-redux';
+import { useEffect } from 'react';
+import { useSelector } from 'react-redux';
 import PropTypes from 'prop-types';
 import NProgress from 'nprogress';
 
-class Spinner extends React.Component {
-  componentDidUpdate(prevProps) {
-    const diff = this.props.pendingTasks - prevProps.pendingTasks;
-    if (diff > 0) {
+const pendingTasksSelector = (state) => state.pendingTasks;
+
+const Spinner = ({ config }) => {
+  const pendingTasks = useSelector(pendingTasksSelector);
```

Comment:
You need to store previous pending tasks count to be able to calculate diff

https://reactjs.org/docs/hooks-faq.html#how-to-get-the-previous-props-or-state